src/server/services/validate-upload.js

```diff
@@ -225,29 +225,43 @@ async function validateRecord ({ upload, record, typeRules: rules, trns }) {
   // check all the rules
   for (const [key, rule] of Object.entries(rules)) {
     // if the rule only applies on different EC codes, skip it
-    if (rule.ecCodes && (!upload.ec_code || rule.ecCodes.indexOf(upload.ec_code) < 0)) {
+    if (rule.ecCodes && (!upload.ec_code || !rule.ecCodes.includes(upload.ec_code))) {
       continue
     }
 
+    // if the field is unset/missing/blank, is that okay?
```

Comment:
Might be useful to expand this comment to mention clearly that a numeric `0` is not an unset value, which is why we aren't using the normal falsy evaluation